prime-dotnet-webapi/Services/EnrolleePaperSubmissionService.cs

```diff
@@ -151,6 +151,15 @@ public async Task UpdateCertificationsAsync(int enrolleeId, IEnumerable<PaperEnr
             await _context.SaveChangesAsync();
         }
 
+        public async Task UpdateDeviceProviderAsync(int enrolleeId, string deviceProviderIdentifier)
+        {
+            var enrollee = await _context.Enrollees
+                .SingleOrDefaultAsync(e => e.Id == enrolleeId);
```

Comment:
If you are calling a `...OrDefaultAsync()` on something and then not checking for null because you assume the object will exist, its more accurate to call `SingleAsync()` since that aligns with your assumption that exactly one exists.

Reply:
Added a null check. The rest of the file still uses `SingleOrDefault` without null checks though, maybe we should have those corrected as well?